### PR TITLE
 Add directories for extended queuejob

### PR DESCRIPTION
**Pull request by dmatch01**

### CODE REVIEW

**Comment by dmatch01** [comment]
@scostache Can you review and add the /lgtm label once you review and approve?

File: deployment/kube-queue-ctrl.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        component: controller
+        tier: control-plane
+        version: second
+    spec:
+      containers:
+      - command:
+        - /bin/sh
+        - -c
+        - touch /tmp/healthy; /opt/kube-queue-ctrl --scheduler-name kube-batchd; rm -f /tmp/healthy
+        image: kubearbitrator/queue-ctrl:v0.1
```
Comment (k82cn) [inline deployment/kube-queue-ctrl.yaml]: we should not use this repos; it's changed to `kubesigs/kube-batch`

File: pkg/apis/controller/.gitignore [hunk 2]
```diff
@@ -0,0 +1,2 @@
+*
+!.gitignore
```
Comment (k82cn) [inline pkg/apis/controller/.gitignore:2]: Why introduce those `.gitignore` files?

File: deployment/kube-queue-ctrl.yaml (hunk 1 above)
Comment (k82cn) [reply deployment/kube-queue-ctrl.yaml]: And where is the dockerfile build this images? we already move them out.

File: pkg/apis/controller/.gitignore (hunk 2 above)
Comment (dmatch01) [reply pkg/apis/controller/.gitignore:2]: Per your past recommendation to submit smaller sized PRs to improve merge consume-ability this first PR submitted for your review creates isolated directories to contain the extended queuejob thus the gitignore files are needed.  As we submit future PRs to add to these directories the gitinore files will be adjusted/removed.

File: deployment/kube-queue-ctrl.yaml (hunk 1 above)
Comment (dmatch01) [reply deployment/kube-queue-ctrl.yaml]: Removed from PR and will make appropriate changes when re-adding to a future PR.

File: pkg/apis/controller/.gitignore (hunk 2 above)
Comment (k82cn) [reply pkg/apis/controller/.gitignore:2]: but .gitignore does not make sense to me.

Comment (dmatch01) [reply pkg/apis/controller/.gitignore:2]: Was you question about the content of the gitignore file?  My understanding to make a directory empty (in the repository) that will checkout after a merge is to create a .gitignore file inside that empty directory that contains those lines.  As we submit future PRs to add to these directories the gitinore files will be adjusted/removed.  If you prefer not to have initial directories empty on the first PR I can add a few of the files that will be within each directory to keep the PR small/med in size and remove the gitignore file.  Please advise.

**Pull request closed without merging**